gcc/rust/typecheck/rust-hir-trait-ref.h

```diff
@@ -380,6 +380,16 @@ class TraitReference
     return item_refs;
   }
 
+  void get_trait_items_and_supers (
+    std::vector<const TraitItemReference *> &result) const
+  {
+    for (const auto &item : item_refs)
+      result.push_back (&item);
+
+    for (const auto &super_trait : super_traits)
+      super_trait->get_trait_items_and_supers (result);
+  }
```

Comment:
I believe this is already a problem on the `master` branch and not necessarily related to this PR, but I'm a bit afraid of the recursive function without a stop condition. The following sort of test cases crash:
```rust
pub trait A: B {}

pub trait B: A {}
```

Reply:
oh good point will fix

Reply:
Let's raise this as a separate issue I have known about this for a while it needs to be caught during resolving a trait.

Reply:
#1591